src/FieldArray.jl

```diff
@@ -4,7 +4,9 @@
 function construct_type(::Type{FA}, x) where {FA <: FieldArray}
     has_size(FA) || error("$FA has no static size!")
     length_match_size(FA, x)
-    return adapt_eltype(FA, x)
+    FA′ = adapt_eltype(FA, x)
+    FA′ === FA && x isa Args && error("the constructor for $FA is missing!")
```

Comment:
```suggestion
    FA′ === FA && x isa Args && error("the constructor for $FA from $x is missing!")
```
In the test `Position1088` does have _a_ constructor, just not the one that is needed, so I think it would be good to have a more clear error message.

Reply:
I tried to expand it as `Args` is not public, now this gives
```julia
julia> Position((1.,2.,3.))
ERROR: The constructor for Position{Float64}(::Float64, ::Float64, ::Float64) is missing!
```
should be more intuitive.

Reply:
Shouldn't it be `Position{Float64}(::typeof(x))`?

Reply:
`x` isa `Args` here, so the `construct_type` was called from `FA(x.args...)`.
Args` was introduced in #1016 to distinguish `SA(x::Tuple)` and `SA(x...)`). It's only used in `construct_type` and would not be passed to the derived constructor.
User should define `Position{T}(x::T, y::T, z::T) where {T}` to fix the constructor missing here.

Reply:
Ah, right, sorry -- I'm not particularly familiar with `FieldArray` stuff.